sync/sync.go

```diff
@@ -28,17 +30,32 @@ type Synchronizer struct {
 	log utils.SimpleLogger
 
 	pendingPollInterval time.Duration
+
+	// metrics
+	opTimers  *prometheus.HistogramVec
+	blockRate prometheus.Counter
 }
 
 func New(bc *blockchain.Blockchain, starkNetData starknetdata.StarknetData,
 	log utils.SimpleLogger, pendingPollInterval time.Duration,
 ) *Synchronizer {
-	return &Synchronizer{
+	s := &Synchronizer{
 		Blockchain:          bc,
 		StarknetData:        starkNetData,
 		log:                 log,
 		pendingPollInterval: pendingPollInterval,
+
+		opTimers: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+			Namespace: "sync",
+			Name:      "timers",
+		}, []string{"op"}),
+		blockRate: prometheus.NewCounter(prometheus.CounterOpts{
```

Comment:
It would be more appropriate to call this `TotalBlocks` since `rate` implies something related to time and this is just a counter.

Reply:
Done